compiler/rustc_errors/src/lib.rs

```diff
@@ -1430,7 +1459,7 @@ pub enum Level {
         /// If this error comes from a lint, don't abort compilation even when abort_if_errors() is called.
         lint: bool,
     },
-    Warning,
+    Warning(Option<LintExpectationId>),
```

Comment:
Probably worth documenting that this `LintExpectationId` is used / can only be set by the `force-warn` flag.